src/main/java/com/timgroup/statsd/UnsendableMessageException.java

```diff
@@ -0,0 +1,20 @@
+package com.timgroup.statsd;
+
+/**
+ * Signals that we've been passed a message that's too long to send
+ *
+ * @author Taylor Schilling
+ */
+
+public class UnsendableMessageException extends RuntimeException {
```

Comment:
Depends on what you're trying to get across. Is it really invalid if it's a validly formed message that's just too big? I have no problem changing the name to whatever since "Unsendable" is certainly vague and doesn't really describe the what or why. `DroppedMessageException` is nice but doesn't solve the "but why?" problem. `MessageBiggerThanWeCareToSendException` is oddly verbose. Maybe "InvalidMessageException" that takes both the message that can't be sent and a short description of why it was invalid.

I'm also overthinking it a bit as it could make sense to have a kind of exception that works for this "larger than we care to send" case as well as the "queue full" case from [#71](https://github.com/DataDog/java-dogstatsd-client/issues/71). Reasonable to have separate explicit exceptions though.

Reply:
I like your last option actually `"InvalidMessageException" that takes both the message that can't be sent and a short description of why it was invalid.`

Reply:
It's nice to have explicit exceptions but if we get into an explosion of exceptions type then it's best to group them (especially if we don't know all of them). It's a judgment call for sure. No right or wrong answers.

Reply:
@schilling Just wanted to touch base with you regarding that above discussion. What are your thoughts at this point? Also, there is now a conflict within `NonBlockingStatsDClientTest.java`, would you mind fixing it so that we can do a final review of the PR?

Reply:
Apologies for the delay. I'll make the change to `InvalidMessageException` as described above and fix up the merge conflict.

Reply:
Ready for additional review.